src/main/java/com/mraof/minestuck/effects/SuspicionEffect.java

```diff
@@ -0,0 +1,136 @@
+	}
+	
+	/**
+	 * Every tick, eject the affected entity's vehicle/passengers and apply a repulsion force from all mobs with this effect within a calculated range.
+	 * @param pLivingEntity the afflicted entity. It will be pushed away from any afflicted entity, and from its ride(r).
+	 * @param pAmplifier    the effect's strength. Used to calculate the repulsion range, is applied an exponentiation smaller than 1 to create a diminishing return.
+	 */
+	@Override
+	public void applyEffectTick(@NotNull LivingEntity pLivingEntity, int pAmplifier)
+	{
+		super.applyEffectTick(pLivingEntity, pAmplifier);
+		
+		if(!hasInitialized)
+		{
+			isFlying = ((pLivingEntity instanceof FlyingMob) || (pLivingEntity instanceof FlyingAnimal));
+			isPushable = !((pLivingEntity instanceof ArmorStand) || (pLivingEntity.getPersistentData().contains("NoAI", 99)));
+			
+			hasInitialized = true;
+		}
+		
+		//If the entity can't be pushed (armor stand/NoAI), there's no point in continuing
+		if(!isPushable)
+		{
+			return;
+		}
+		
+		if(pLivingEntity.getVehicle() != null || pLivingEntity.getPassengers().size() > 0)
+		{
+			pLivingEntity.ejectPassengers();
+			pLivingEntity.dismountTo(pLivingEntity.getX(), pLivingEntity.getY(), pLivingEntity.getZ());
+		}
+		
+		double range = rangeScale * Math.pow(pAmplifier, 0.6);
+		
+		if(pLivingEntity.isOnGround() || isFlying)
+		{
+			for(LivingEntity otherEntity : pLivingEntity.level.getEntitiesOfClass(LivingEntity.class, pLivingEntity.getBoundingBox().inflate(range, 1, range)))
+			{
+				if(otherEntity != pLivingEntity && otherEntity.hasEffect(this))
+				{
+					//Push direction vector, from other to self
+					double pushX = pLivingEntity.getX() - otherEntity.getX();
+					double pushZ = pLivingEntity.getZ() - otherEntity.getZ();
+					double pushMagnitude = Math.sqrt((pushX * pushX + pushZ * pushZ));
+					pushX /= pushMagnitude;
+					pushZ /= pushMagnitude;
+					
+					//Push is stronger the closest each entity is
+					double pushForce = 1 / (Math.max(pushMagnitude, 1));
+					pLivingEntity.push((pushX * pushForce), 0, (pushZ * pushForce));
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Cancels out attempts to ride (or be ridden by) an affected entity
+	 */
+	@SubscribeEvent
+	public void onMount(EntityMountEvent event)
+	{
+		if(isPushable)
+		{
+			Entity entityBeingMounted = event.getEntityBeingMounted();
+			Entity mountEntity = event.getEntityMounting();
+			
+			if(((LivingEntity) entityBeingMounted).hasEffect(this) || ((LivingEntity) mountEntity).hasEffect(this))
+				event.setCanceled(true);
+		}
+	}
+	
+	/**
+	 * Cancels out attempts to call for reinforcements.
+	 * Examples: {@link net.minecraft.world.entity.monster.Zombie#hurt(DamageSource, float)}, {@link net.minecraft.world.entity.monster.ZombifiedPiglin#hurt(DamageSource, float)})
+	 */
+	@SubscribeEvent
+	public void summonAttempt(ZombieEvent.SummonAidEvent event)
+	{
+		if(event.getEntity().hasEffect(this)) event.setCanceled(true);
+	}
```

Comment:
The event handlers does not appear to be properly registered. Try changing them to be static instead: https://cdn.discordapp.com/attachments/437002009216942086/558994977430962196/eventsubscriber.png

Reply:
And I had changed them before because I thought they weren't working because they were static. Fixed.